Store/StoreCacheTableUpdater.php

```diff
@@ -8,9 +8,14 @@
  *
  * @copyright 2006-2016 silverorange
  * @license   http://www.gnu.org/copyleft/lesser.html LGPL License 2.1
+ *
+ * @property StoreCommandLineConfigModule $config
+ * @property SiteDatabaseModule           $database
  */
 class StoreCacheTableUpdater extends SiteCommandLineApplication
 {
+    public MDB2_Driver_Common $db;
```

Comment:
[Above](https://github.com/silverorange/store/pull/185/files#diff-64e9df13afbf2261ef8acedac613522d4759acc912ef6682906dddbcba1eed25R17), we set the type with a docblock comment.  The worry I have with explicitly typing the property would be that we run into the "access before setting" issue ... in which case it needs to be nullable (and then we need to do null checks throughout the code).

Any concerns on your end?

Reply:
I had that concern as well. I did test the cache table updater and it runs correctly. I *think* if we ever had a access before setting issue in the code, we'd also have a null-dereference issue as none of the code checks for null on this property.